lib/go/edgecontext/backend.go

```diff
@@ -0,0 +1,126 @@
+package edgecontext
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/reddit/baseplate.go/ecinterface"
+	v0 "github.com/reddit/edgecontext/lib/go/internal/v0"
+)
+
+func init() {
+	SetBackend(defaultBackend())
+}
+
+var backend atomic.Pointer[Backend]
+
+type Backend struct {
+	SetEdgeContext func(context.Context, *EdgeRequestContext) context.Context
+	GetEdgeContext func(context.Context) (*EdgeRequestContext, bool)
```

Comment:
🔕 Since it's already in an `edgecontext` package and has types with that in the signature, maybe we can drop that from the name?

e.g. Load/Store or Get/Set or Encode/Decode?